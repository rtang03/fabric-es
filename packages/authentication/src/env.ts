import { config } from 'dotenv';
import { resolve } from 'path';

if (process.env.NODE_ENV === 'production') config({ path: './.env' });
else config({ path: resolve(__dirname, '../.env.dev') });
