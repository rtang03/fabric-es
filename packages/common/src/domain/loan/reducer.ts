import { Loan, LoanEvent, LoanStatus } from '.';

export const loanReducer = (
  history: LoanEvent[],
  initialState?: Loan
): Loan => {
  const reducer = (loan: Loan, event: LoanEvent): Loan => {
    switch (event.type) {
      case 'applied':
        return {
          loanId: event.payload.loanId,
          status: LoanStatus[event.type],
          ownerId: event.payload.userId,
          timestamp: event.payload.timestamp,
          reference: null,
          loanProductId: null
        };
      case 'cancelled':
      case 'approved':
      case 'returned':
      case 'rejected':
      case 'expired':
        // loan.status = LoanStatus[event.type];
        return {
          ...loan,
          status: LoanStatus[event.type]
        };
      case 'LoanReferenceDefined':
        return {
          ...loan,
          reference: event.payload.reference
        };
      case 'LoanDescriptionDefined':
        return {
          ...loan,
          description: event.payload.description
        };
      case 'LoanProductDefined':
        return {
          ...loan,
          loanProductId: event.payload.loanProductId
        };
    }
  };

  return history.reduce(reducer, initialState);
};