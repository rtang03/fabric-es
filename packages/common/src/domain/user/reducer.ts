import { User, UserEvent } from '.';

/**
 * User Reducer
 * @param history
 * @param initialState
 */
export function userReducer(history: UserEvent[], initialState?: User): User {
  const reducer = (user: User, event: UserEvent): User => {
    switch (event.type) {
      case 'UserCreated':
        return {
          userId: event.payload.userId,
          name: event.payload.name,
          mergedUserIds: event.payload.mergedUserIds
        };
      case 'ReviewInvitationDeclined':
      case 'ReviewInvitationExpired':
        return {
          ...user
        };
      default:
        return user;
    }
  };

  return history.reduce(reducer, initialState);
}
