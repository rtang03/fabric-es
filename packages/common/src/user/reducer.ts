import { User, UserEvents } from '.';

/**
 * User Reducer
 * @param history
 * @param initialState
 */
export function userReducer(history: UserEvents[], initialState?: User): User {
  const reducer = (user: User, event: UserEvents): User => {
    switch (event.type) {
      case 'UserCreated':
        return {
          userId: event.payload.userId,
          name: event.payload.name,
          mergedUserIds: event.payload.mergedUserIds
        };
      case 'ReviewInvitationDeclined':
      case 'ReviewInvitationExpired':
        return {
          ...user
        };
      default:
        return user;
    }
  };

  return history.reduce(reducer, initialState);
}
