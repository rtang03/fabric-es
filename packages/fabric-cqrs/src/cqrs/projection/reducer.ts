/**
 * @packageDocumentation
 * @hidden
 */
import { Reducer } from 'redux';
import { ActionHandler, initialState, State } from '../../types';
import { getErrorActionHandler, getReducer, getSuccessActionHandler } from '../../store/utils';
import { action } from './action';

const actionHandler: ActionHandler = {
  [action.UPSERT_MANY_SUCCESS]: getSuccessActionHandler(action.UPSERT_MANY_SUCCESS),
  [action.UPSERT_SUCCESS]: getSuccessActionHandler(action.UPSERT_SUCCESS),
  [action.FIND_SUCCESS]: getSuccessActionHandler(action.FIND_SUCCESS)
};

export const reducer: Reducer<State> = getReducer(initialState, actionHandler);
