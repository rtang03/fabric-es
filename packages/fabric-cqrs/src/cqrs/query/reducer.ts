/**
 * @packageDocumentation
 * @hidden
 */
import { Reducer } from 'redux';
import { getErrorActionHandler, getReducer, getSuccessActionHandler } from '../../store/utils';
import { ActionHandler, initialState, State } from '../../types';
import { action } from './action';

const actionHandler: ActionHandler = {
  [action.MERGE_SUCCESS]: getSuccessActionHandler(action.MERGE_SUCCESS),
  [action.MERGE_ERROR]: getErrorActionHandler(action.MERGE_ERROR),
  [action.MERGE_BATCH_SUCCESS]: getSuccessActionHandler(action.MERGE_BATCH_SUCCESS),
  [action.MERGE_BATCH_ERROR]: getErrorActionHandler(action.MERGE_BATCH_ERROR),
  [action.DELETE_SUCCESS]: getSuccessActionHandler(action.DELETE_SUCCESS),
  [action.DELETE_ERROR]: getErrorActionHandler(action.DELETE_ERROR),
  [action.QUERY_SUCCESS]: getSuccessActionHandler(action.QUERY_SUCCESS),
  [action.QUERY_ERROR]: getErrorActionHandler(action.QUERY_ERROR)
};

export const reducer: Reducer<State> = getReducer(initialState, actionHandler);
