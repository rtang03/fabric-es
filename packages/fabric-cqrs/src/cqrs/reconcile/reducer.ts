import { Reducer } from 'redux';
import { ActionHandler, initialState, State } from '../../types';
import {
  getErrorActionHandler,
  getReducer,
  getSuccessActionHandler
} from '../utils';
import { action } from './action';

const actionHandler: ActionHandler = {
  [action.RECONCILE_SUCCESS]: getSuccessActionHandler(action.RECONCILE_SUCCESS),
  [action.RECONCILE_ERROR]: getErrorActionHandler(action.RECONCILE_ERROR)
};

export const reducer: Reducer<State> = getReducer(initialState, actionHandler);
