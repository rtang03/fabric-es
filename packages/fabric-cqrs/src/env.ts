import { resolve } from 'path';
import { config } from 'dotenv';

if (process.env.NODE_ENV !== ' production') config({ path: resolve(__dirname, '../.env.dev') });
