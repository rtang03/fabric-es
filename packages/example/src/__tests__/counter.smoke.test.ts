import {
  bootstrapNetwork,
  createPeer,
  Peer,
  Repository
} from '@espresso/fabric-cqrs';
import { Gateway, Network } from 'fabric-network';
import { pick } from 'lodash';
import '../env';
import { Counter, CounterEvent, reducer } from '../reducer';

let network: Network;
let gateway: Gateway;
let peer: Peer;
let repo: Repository<Counter, CounterEvent>;
const entityName = 'ex_counter';
const enrollmentId = `counter_ex_test${Math.floor(Math.random() * 10000)}`;

beforeAll(async () => {
  const context = await bootstrapNetwork({
    enrollmentId
  });
  network = context.network;
  gateway = context.gateway;
  peer = createPeer({
    ...context,
    defaultEntityName: entityName,
    defaultReducer: reducer,
    collection: 'Org1PrivateDetails'
  });
  await peer.subscribeHub();
  repo = peer.getRepository<Counter, CounterEvent>({ entityName, reducer });
});

afterAll(async () => {
  peer.unsubscribeHub();
  gateway.disconnect();
});

/**
 * This is smoke test only, to validate gateway connection
 */
describe('Example Smoke Test', () => {
  beforeAll(async () => {
    await repo
      .deleteByEntityId(enrollmentId)
      .then(() => true)
      .catch(() => true);
    await repo
      .deleteByEntityName_query()
      .then(({ status }) => status)
      .then(res => expect(res).toEqual('all records deleted successfully'));
  });

  it('should Add', async () =>
    repo
      .create({ enrollmentId, id: enrollmentId })
      .save([{ type: 'ADD' }])
      .then(result => pick(result, 'version', 'entityName', 'events'))
      .then(result => expect(result).toMatchSnapshot()));
});
