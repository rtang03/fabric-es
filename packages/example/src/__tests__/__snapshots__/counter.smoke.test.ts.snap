// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example Smoke Test should Add 1`] = `
Object {
  "entityName": "ex_counter",
  "events": Array [
    Object {
      "type": "ADD",
    },
  ],
  "version": 0,
}
`;
